dot/rpc/modules/chain_test.go

```diff
@@ -2,3 +2,529 @@
+			Number:         "0x00",
+			StateRoot:      "0x0000000000000000000000000000000000000000000000000000000000000000",
+			ExtrinsicsRoot: "0x0000000000000000000000000000000000000000000000000000000000000000",
+			Digest:         ChainBlockHeaderDigest{},
+		},
+		Body: nil,
+	}}
+
+	expBlock := &ChainBlockResponse{Block: ChainBlock{
+		Header: ChainBlockHeaderResponse{
+			ParentHash:     "0x0000000000000000000000000000000000000000000000000000000000000000",
+			Number:         "0x00",
+			StateRoot:      "0x0000000000000000000000000000000000000000000000000000000000000000",
+			ExtrinsicsRoot: "0x0000000000000000000000000000000000000000000000000000000000000000",
+			Digest:         ChainBlockHeaderDigest{},
+		},
+		Body: []string{"0x0401"},
+	}}
+
+	chainModule := NewChainModule(mockBlockAPI)
+	type fields struct {
+		blockAPI BlockAPI
+	}
+	type args struct {
+		r   *http.Request
+		req *ChainHashRequest
+		res *ChainBlockResponse
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		exp     *ChainBlockResponse
+		wantErr bool
+		err     error
+	}{
+		{
+			name: "GetBlock OK",
+			fields: fields{
+				chainModule.blockAPI,
+			},
+			args: args{
+				req: &ChainHashRequest{},
+			},
+			exp: expEmptyBlock,
+		},
+		{
+			name: "GetBlockByHash Err",
+			fields: fields{
+				mockBlockAPIGetHashErr,
+			},
+			args: args{
+				req: &ChainHashRequest{&testHash},
+			},
+			wantErr: true,
+			err:     errors.New("GetJustification error"),
+		},
+		{
+			name: "GetBlock with body OK",
+			fields: fields{
+				mockBlockAPIWithBody,
+			},
+			args: args{
+				req: &ChainHashRequest{&testHash},
+			},
+			exp: expBlock,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res := ChainBlockResponse{}
+			tt.args.res = &res
+			cm := &ChainModule{
+				blockAPI: tt.fields.blockAPI,
+			}
+			err := cm.GetBlock(tt.args.r, tt.args.req, tt.args.res)
+			if tt.wantErr {
+				if assert.NotNil(t, err) {
+					assert.EqualError(t, err, tt.err.Error())
+				}
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.exp, tt.args.res)
+			}
+		})
+	}
+}
+
+func TestChainModule_GetBlockHash(t *testing.T) {
+	testHash := common.NewHash([]byte{0x01, 0x02})
+	i := []interface{}{"a"}
+
+	mockBlockAPI := new(apimocks.BlockAPI)
+	mockBlockAPI.On("BestBlockHash").Return(testHash, nil)
+	mockBlockAPI.On("GetBlockHash", new(big.Int).SetInt64(int64(21))).Return(testHash, nil)
+
+	mockBlockAPIErr := new(apimocks.BlockAPI)
+	mockBlockAPIErr.On("BestBlockHash").Return(testHash, nil)
+	mockBlockAPIErr.On("GetBlockHash", new(big.Int).SetInt64(int64(21))).Return(nil, errors.New("GetBlockHash Error"))
+
+	expRes := ChainHashResponse(testHash.String())
+	type fields struct {
+		blockAPI BlockAPI
+	}
+	type args struct {
+		r   *http.Request
+		req *ChainBlockNumberRequest
+		res *ChainHashResponse
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+		err     error
```

Comment:
```suggestion
		expErr     error
```

Reply:
I actually use `err` for the expected error :smile: I guess each build their own little naming conventions here and there